sdk/python/lib/pulumi/output.py

```diff
@@ -253,27 +253,32 @@ def from_input(val: Input[T]) -> 'Output[T]':
         if isinstance(val, dict):
             # Since Output.all works on lists early, serialize this dictionary into a list of lists first.
             # Once we have a output of the list of properties, we can use an apply to re-hydrate it back into a dict.
-            transformed_items = [[k, Output.from_input(v)] for k, v in val.items()]
-            return Output.all(*transformed_items).apply(lambda props: {k: v for k, v in props}, True)
+            dict_items = [[k, Output.from_input(v)] for k, v in val.items()]
+            # type checker doesn't like returing a Dict in the apply callback
+            fn = cast(Callable[[List[Any]], T], lambda props: {k: v for k, v in props}) # pylint: disable=unnecessary-comprehension
+            return Output.all(*dict_items).apply(fn, True)
 
         if isinstance(val, list):
-            transformed_items = [Output.from_input(v) for v in val]
-            return Output.all(*transformed_items)
+            list_items: List[Union[Any, Awaitable[Any], Output[Any]]] = [Output.from_input(v) for v in val]
+            # invariant: http://mypy.readthedocs.io/en/latest/common_issues.html#variance
+            output: Output[T] = cast(Output[T], Output.all(*list(list_items))) # type: ignore
+            return output
 
         # If it's not an output, list, or dict, it must be known and not secret
-        is_known_fut = asyncio.Future()
-        is_secret_fut = asyncio.Future()
+        is_known_fut: asyncio.Future[bool] = asyncio.Future()
+        is_secret_fut: asyncio.Future[bool] = asyncio.Future()
         is_known_fut.set_result(True)
         is_secret_fut.set_result(False)
 
         # Is it awaitable? If so, schedule it for execution and use the resulting future
         # as the value future for a new output.
         if isawaitable(val):
-            promise_output = Output(set(), asyncio.ensure_future(val), is_known_fut, is_secret_fut)
+            val_fut = cast(asyncio.Future, val)
+            promise_output = Output(set(), asyncio.ensure_future(val_fut), is_known_fut, is_secret_fut)
             return promise_output.apply(Output.from_input, True)
 
         # Is it a prompt value? Set up a new resolved future and use that as the value future.
-        value_fut = asyncio.Future()
+        value_fut: asyncio.Future = asyncio.Future()
```

Comment:
what's the value of this annotation?

Reply:
`mypy` actually errors if you don't have it.  But if we are going to have it (to appease mypy) we may as well be explicit that this is a synonym for `asyncio.Future[Any]`.

Reply:
Changed.

Reply:
Added the explicit `any` where appropriate.